pkg/expression/builtin_compare.go

```diff
@@ -1586,6 +1602,28 @@ func (c *compareFunctionClass) refineArgs(ctx BuildContext, args []Expression) (
 		return nil, err
 	}
 
+	if c.op == opcode.NullEQ && arg0IsCon && arg0.DeferredExpr == nil && !arg1IsCon && arg1Type.GetType() == mysql.TypeDuration {
+		isNull, err := castToDurationIsNull(ctx, arg0)
+		if err != nil {
+			return nil, err
+		}
+
+		if isNull {
+			return []Expression{NewZero(), NewOne()}, nil
+		}
+	}
+
+	if c.op == opcode.NullEQ && arg1IsCon && arg1.DeferredExpr == nil && !arg0IsCon && arg0Type.GetType() == mysql.TypeDuration {
+		isNull, err := castToDurationIsNull(ctx, arg1)
+		if err != nil {
+			return nil, err
+		}
+
+		if isNull {
+			return []Expression{NewZero(), NewOne()}, nil
+		}
+	}
```

Comment:
Add some comments for why we're add this if block

Reply:
> Add some comments for why we're add this if block

done